src/lib/Zikula/Bundle/CoreBundle/EventListener/ThemeListener.php

```diff
@@ -161,16 +164,63 @@ public function setUpThemePathOverrides(FilterControllerEvent $event)
         }
     }
 
+    /**
+     * Read the controller annotations and change theme if the annotation indicate that need
+     * @param FilterControllerEvent $event
+     */
+    public function readControllerAnnotations(FilterControllerEvent $event)
+    {
+        if (!$event->isMasterRequest()) {
+            // prevents calling this for controller usage within a template or elsewhere
+            return;
+        }
+        $controller = $event->getController();
+        list($controller, $method) = $controller;
+        $this->changeThemeByAnnotation($controller, $method);
+    }
+
+    /**
+     * Change a theme based on the annotation
+     * @param $controller
+     * @param $method
+     * @return array|bool|string
+     */
+    public function changeThemeByAnnotation($controller, $method)
```

Comment:
I've no real opinion about this. Maybe we need a general distinction of what comes into the core bundle and what not. My feeling says put it into the bundle but I have no reasoning.